packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -256,6 +256,13 @@ class Context {
 
   // Add the vmService publisher Bonjour service to the produced app bundle Info.plist.
   void addVmServiceBonjourService() {
+    // Skip adding Bonjour service settings when DISABLE_PORT_PUBLICATION is YES.
+    // These settings are not needed if port publication is disabled.
+    if (environment['DISABLE_PORT_PUBLICATION'] != null &&
```

Comment:
do we need the != null comparison? would this not be equivalent to only checking if the value == 'YES'?

Reply:
Oh good point, I'l update